File: Java-OOP-Advanced/Problems Course Instance 2016_2017/src/Lesson09ObjectCommunicationAndEvents/Lab/commands/AttackCommand.java
package Lesson09ObjectCommunicationAndEvents.Lab.commands;

import Lesson09ObjectCommunicationAndEvents.Lab.interfaces.Attacker;
import Lesson09ObjectCommunicationAndEvents.Lab.interfaces.Command;

public class AttackCommand implements Command {

    private Attacker attacker;

    public AttackCommand(Attacker attacker) {
        this.attacker = attacker;
    }

    @Override
    public void execute() {
        this.attacker.attack();
    }
}

File: PHP-Web-PHP-Basics/2017 Spring/Resources/7.3. PHP-Fundamentals-OOP-Advanced-Interfaces-And-Abstraction-Demos/Core/Commands/AttackCommand.php
<?php


namespace Core\Commands;


class AttackCommand extends CommandAbstract
{

    public function execute(array $args = []): string
    {
        array_shift($args);
        $attackerName = array_shift($args);
        $defenderName = array_shift($args);

        $attacker = $this->galaxy->getShip($attackerName);
        $defender = $this->galaxy->getShip($defenderName);

        if (!$attacker->isAlive() || !$defender->isAlive()) {
            throw new \Exception("Ship is destroyed");
        }

        if ($attacker->getStarSystem() != $defender->getStarSystem()) {
            throw new \Exception("No such ship in star system");
        }

        $attacker->attack($defender);

        $output = "$attackerName attacked $defenderName";

        if (!$defender->isAlive()) {
            $output .= PHP_EOL . "$defenderName has been destroyed";
        }

        return $output;
    }
}